src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/web/WorkflowRunController.kt

```diff
@@ -110,24 +105,35 @@ class WorkflowRunController(
         return passThroughResponse(response)
     }
 
+    @NoCoverage
     internal data class WorkflowRunStatusResponse(
         @SerializedName(value = "workflow_key")
         val key: String,
-        val status: String
+        val status: String,
+        val reports: List<WorkflowRunStatusResponseReport>
     )
+    {
+        @NoCoverage
+        data class WorkflowRunStatusResponseReport(
+            val key: String,
+            val status: String,
+            val version: String?
+        )
+    }
 
-    fun getWorkflowRunStatus(): String
+    fun getWorkflowRunStatus(): WorkflowRunStatus
     {
         val key = context.params(":key")
-        val response = orderlyServerAPI.get(
-            "/v1/workflow/$key/status/",
-            emptyMap()
-        )
-        if (response.statusCode == HTTP_OK)
-        {
-            val workflowRunStatusResponse = response.data(WorkflowRunStatusResponse::class.java)
-            workflowRunRepository.updateWorkflowRun(workflowRunStatusResponse.key, workflowRunStatusResponse.status)
-        }
-        return passThroughResponse(response)
+        val response = orderlyServerAPI
+            .throwOnError()
```

Comment:
This `throwOnError`is going to change the error response behaviour isn't it? Rather than just passing through the Orderly server error response, it just generates an error with the code and url I think? This is probably ok but might lose error information?

Reply:
> Great idea on the annotation - maybe add a note to the README on the intended usage (i.e. what it's _not_ to be used for)?

Done!

Reply:
> This `throwOnError`is going to change the error response behaviour isn't it? Rather than just passing through the Orderly server error response, it just generates an error with the code and url I think? This is probably ok but might lose error information?

Yes, good point. I hadn't considered that, but I think this approach is still valid. `throwOnError()` is obviously designed for this purpose and usage here is consistent with other cases. I think there is indeed an argument for modifying its implementation to include further information about the root cause, but that would in turn change the response from other endpoints. And I'm not concerned about changing the behaviour of this one because a) it's web only right now and b) it's new anyway.